### PR TITLE
NetCare Hospital App

### PR DESCRIPTION
**Pull request by Munyabelden**
### In this PR we

- Created  an App call NetCare app
- Created a database called Final_capstone
- Added 3 tables  Users, Doctors, Consultation and City
- Added Logic so that User is authenticated that is can sign up and login
- User can add data and book a consultation
- Created a navigation ith links redirecting you around the App

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,40 +77,79 @@ To get a local copy up and running, follow these steps.
...
+  rspec
+```
+### Deployment
 
-Run: rspec to run tests
+You can deploy this project using:
+
+- [ ] Use render or any other hosting of your choosing
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (mcrd25) [inline README.md:152]: [OPTIONAL] Consider adding instructions on how to deploy, especially sicne you don't have a live demo link. 

- Also, please add a live demo link at some point

File: app/models/doctor.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Doctor < ApplicationRecord
+  has_many :consultations
+
+  validates :name, presence: true
```
Comment (mcrd25) [inline app/models/doctor.rb:4]: Is presence the only thing you can validate? What about length [OPTIONAL]

File: app/models/consultation.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Consultation < ApplicationRecord
+  belongs_to :user
+  belongs_to :doctor
+
+  validates :duration, presence: true, numericality: { only_integer: true, greater_than: 0 }
+  validates :city, presence: true
+  validates :date, presence: true
```
Comment (mcrd25) [inline app/models/consultation.rb:7]: [OPTIONAL]
Is presence the only thing yo ucan valdiate here?

File: app/controllers/api/v1/consultations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  def update
+    if @consultation.update(consultation_params)
+      render json: @consultation
+    else
+      render json: @consultation.errors, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @consultation.destroy
```
Comment (mcrd25) [inline app/controllers/api/v1/consultations_controller.rb:34]: What happens if it does not destroy successfully?
What message is displayed?

File: app/controllers/api/v1/doctors_controller.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      render json: {
+        error: 'Error creating service...'
+      }
+    end
+  end
+
+  def destroy
+    @doctor = Doctor.find(params[:id])
+    @doctor.destroy
+    head :no_content
```
Comment (mcrd25) [inline app/controllers/api/v1/doctors_controller.rb:34]: What happens if there is an error when performs the first two tasks? Don't foget your error handling

File: README.md [hunk 6]
```diff
@@ -26,11 +28,17 @@
...
+# 📖 [NetCare App] <a name="The Healthnet App helps you to find doctors and book appointment for various healthcare consultations."></a>
+
 
+**[NetCare App]** is based on Ruby on Rails base app for that allows petient and doctors to get in touch and schedule consultations.
 
-**[Healthnet App]** is based on Ruby Language and Rails framework.
+ [Kanban Board](https://github.com/users/Munyabelden/projects/4/views/1?layout=board)
+ 
+ <img src="https://user-images.githubusercontent.com/104892694/254542615-4d46c327-0f12-46a4-a855-8dedf4dc7bc7.png" alt="img"/>
+ Below is the link to the database diagram
```
Comment (mcrd25) [inline README.md:39]: You should probably reference your front-end repo at some point

**Review by mcrd25 — COMMENTED** [review]
Hi Team,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Routes work well [backend] 👍 
- API works well 👍 
- Met minimum requirements 👍 

## Required Changes ♻️
_Check the comments under the review._
- I am not seeing reference to front-end PR, please add this so it can be reviewed. I left some general things to consider before next review in the front-end below
- you are missing documentation. Consider using the `rswag` gem : https://github.com/rswag/rswag
  You may also use: 
   - swagger-docs: https://github.com/richhollis/swagger-docs
   - swagger-ui_rails2: https://github.com/ryansyl/swagger_ui_rails2
- Not seeing unit tests, and in general, the tests are slightly underwhelming ( i only see one set of tests)
- 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [OPTIONAL] Should always have positive and negative tests to ensure it works the way you expect it. Eg. What do you expect the output to be when it's not valid or when valid etc.
Consider making all your tests more extensive, not just this one. For this simple application, it may not catch a possible crippling bug, but in bigger applications, it definitely can. Consider this for all projects moving forward.
- [OPTIONAL]consider removing the `test` directory since you are using `rspec` for testing and not native testing.
- [OPTIONAL] Consider using the `timecop` gem to freeze time when running tests. It can help avoid errors: https://github.com/travisjeffery/timecop
#### FRONT-END (Things to consider since i did not get to review)
- Don't forget that the application needs to look exactly like design with exception of photos, titles and descriptions
* You may not have tests for your Front-End code - this is something we expect to be implemented for the final evaluation. Here are some resources you can use:
   *  https://reactjs.org/docs/testing.html
   * https://jestjs.io/docs/tutorial-react
   * https://www.freecodecamp.org/news/testing-react-hooks/
- See where you can make code better by following best practices: See the following articles
   - https://www.freecodecamp.org/news/best-practices-for-react/
   - https://betterprogramming.pub/21-best-practices-for-a-clean-react-project-df788a682fb
- Particularly, see where you can make your code resuable

**Hope all is well, and that my comments were helpful. 😸**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
_Also, you do not have to implement all requested changes, as this is up to you in the end, but I strongly recommend you implement them._

**Comment by Munyabelden** [comment]
Thanks for the feedback @mcrd25

**Pull request merged**